conductor/src/lib.rs

```diff
@@ -189,6 +210,25 @@ pub async fn delete_namespace(client: Client, name: &str) -> Result<(), Conducto
         .delete(name, &params)
         .await
         .map_err(ConductorError::KubeError);
+
+    // Watch for deletion
+    let wp = WatchParams {
+        field_selector: Some(format!("metadata.name={}", name)),
+        ..WatchParams::default()
+    };
+    let stream = ns_api
+        .watch(&wp, "0")
+        .await
+        .map_err(ConductorError::KubeError)?;
+
+    let mut pinned_stream = Box::pin(stream);
+
+    while let Some(status) = pinned_stream.try_next().await? {
+        if let WatchEvent::Deleted(_) = status {
+            info!("Namespace {} deleted", name);
+            return Ok(());
+        }
+    }
```

Comment:
Adding a `WatchEvent` to conductor.  This will ensure that the function will not return unless there is an error or if the `Namespace` is actually deleted.